File: reddio-csharp/reddio.api/V1/Rest/GetBalancesMessage.cs
using Newtonsoft.Json;

namespace Reddio.Api.V1.Rest
{
    public class GetBalancesMessage
    {
        public GetBalancesMessage(string starkKey, string contractAddress)
        {
            StarkKey = starkKey;
            ContractAddress = contractAddress;
        }

        public GetBalancesMessage(string starkKey)
        {
            StarkKey = starkKey;
        }

        public GetBalancesMessage()
        {
        }

        [JsonProperty("stark_key")] public string StarkKey { get; set; }
        [JsonProperty("contract_address")] public string ContractAddress { get; set; }
    }
}

File: reddio-java/reddio-api/src/main/java/com/reddio/api/v3/rest/GetBalancesMessage.java
package com.reddio.api.v3.rest;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor(staticName = "of")
public class GetBalancesMessage {

    @JsonProperty("stark_key")
    private String starkKey;

    @JsonProperty("page")
    private Long page;

    @JsonProperty("limit")
    private Long limit;

    @JsonProperty("contract_address")
    private String contractAddress;

    @JsonProperty("type")
    private String type;
}
